app/models/job.py

```diff
@@ -205,6 +212,13 @@ def cancel(self):
     def original_file_name_without_extention(self):
         return Path(self.original_file_name).stem
 
+    @property
+    def service_data_retention_days(self):
+        return self._get_service_data_retention_days()
+
+    def _get_service_data_retention_days(self):
+        return service_api_client.get_service_data_retention(self.service)[0]["days_of_retention"]
+
```

Comment:
```suggestion
    @cached_property
    def service_data_retention_days(self):
        return service_api_client.get_service_data_retention(self.service)[0]["days_of_retention"]

```

- you don’t need a separate private method here
- use `cached_property` to prevent hitting the API multiple times for the same data